packages/layout/src/Grid.ts

```diff
@@ -488,7 +490,12 @@ export class Grid extends HTMLWidget {
                     this.postSelectionChange();
                 }
             } else {
-                this._selectionBag.set([selectionObj]);
+                const selected = this._selectionBag.get();
+                if (selected[0] && selected[0]._id === selectionObj._id) {
```

Comment:
Don't you need to check all the currently selected items to see if the click is already selected?

Reply:
No, because this selection clearing feature only applies if a single element is selected. If two cells were selected and a user clicks on one of them, the expected behavior would be to select the one they clicked rather than deselecting both of them. Though now I'm realizing that first condition should be `selected.length === 1` rather than `selected[0]`